packages/api/src/derives/cennzx/cennzxTrade.e2e.ts

```diff
@@ -1,496 +1,492 @@
+                  expect(liquidity.gtn(0)).toBeTruthy();
+                  done();
+                }
+              }
+            }
+          });
+      });
+    });
+
+    it("Add liquidity for tradeAsset 'B' to the pool", async done => {
+      // Amount of test asset - 'B' to create
+      const initialIssuance = 900_000_000_00;
+      const owner = api.registry.createType('Owner', 0); // Owner type is enum with 0 as none/null
+      const permissions = api.registry.createType('PermissionsV1', { update: owner, mint: owner, burn: owner });
+      const option = { initialIssuance, permissions };
+      const assetOption: AssetOptions = api.registry.createType('AssetOptions', option);
+      const assetInfo: AssetInfo = api.registry.createType('AssetInfo', { symbol: 'B', decimalPlaces: 18 });
+      const createAssetTx2 = api.tx.genericAsset.create(bob.address, assetOption, assetInfo);
+      // when the new asset is created it will have this ID.
+      assetB = await api.query.genericAsset.nextAssetId();
+      // Create new asset
+      const assetCreated = new Promise(async (resolve, reject) => {
+        await api.tx.sudo
+          .sudo(createAssetTx2)
+          .signAndSend(sudoKeypair, { nonce: nonce++ }, async ({ status }) => (status.isInBlock ? resolve() : null));
+      });
+      // AddLiquidity for the 'B' asset
+      assetCreated.then(async () => {
+        const assetBalance = await api.query.genericAsset.freeBalance(assetB, bob.address);
+        const investmentAmount = new BN(assetBalance.toString()).divn(2);
+        const coreAmount = investmentAmount;
+        const minLiquidity = 1;
+        await api.tx.cennzx
+          .addLiquidity(assetB, minLiquidity, investmentAmount, coreAmount)
+          .signAndSend(bob, async ({ events, status }) => {
+            if (status.isFinalized) {
+              for (const { event } of events) {
+                if (event.method === 'AddLiquidity') {
+                  const [account, coreInvestAmount, assetIdFromChain, targetInvestAmount] = event.data;
+                  expect(account.toString()).toEqual(bob.address);
+                  expect(assetIdFromChain.toString()).toEqual(assetB.toString());
+                  expect(coreInvestAmount.toString()).toEqual(coreAmount.toString());
+                  expect(new BN(targetInvestAmount).lte(new BN(investmentAmount))).toBeTruthy();
+                  const liquidity = await api.derive.cennzx.liquidityBalance(assetB, bob.address);
+                  expect(liquidity.gtn(0)).toBeTruthy();
+                  done();
+                }
+              }
+            }
+          });
+      });
+    });
+  });
+
+  describe('Trading Operations', () => {
+    it('Can trade from asset to core for exact core asset amount', async done => {
+      const recipient = null; // in which case the caller receives the exchanged amount
+      const assetToSell = 17000;
```

Comment:
Is there a reason it's not assigned from assetA?

Reply:
no.. was testing these test cases one by one so had hardcode... removed all the other, but this one was left..thanks.. will update.. n test once more..

Reply:
done and tested..